src/components/NewWorkspaceModal.ts

```diff
@@ -192,6 +199,42 @@ const NewWorkspaceModal = withDisplayName(
           }
         })();
 
+        if (getProjectCloudPlatform() === 'AZURE' && waitForServices?.wds) {
+          // WDS takes some time to start up, so there's no need to immediately start checking if it's running.
+          await delay(30000);
+
+          // Wait for the WDS app to be running.
+          const wds = await Utils.poll(
+            async () => {
+              const workspaceApps: ListAppResponse[] = await Ajax().Apps.listAppsV2(createdWorkspace.workspaceId);
+              const wdsApp = resolveWdsApp(workspaceApps);
+              if (wdsApp?.status === 'RUNNING') {
+                return { shouldContinue: false, result: wdsApp };
+              }
+              if (wdsApp?.status === 'ERROR') {
+                throw new Error('Failed to provision data services for new workspace.');
+              }
+              return { shouldContinue: true, result: null };
+            },
+            15000,
+            true
+          );
+
+          // Wait for the default WDS instance to exist.
+          const proxyUrl = wds!.proxyUrls.wds;
+          await Utils.poll(
+            async () => {
+              const instances: string[] = await Ajax().WorkspaceData.listInstances(proxyUrl);
+              if (instances.includes(createdWorkspace.workspaceId)) {
+                return { shouldContinue: false, result: true };
+              }
+              return { shouldContinue: true, result: false };
+            },
+            5000,
+            true
+          );
+        }
```

Comment:
feels like a lot of this logic is similar to what we have on the Data tab. Is there anything reusable here?

Reply:
Probably, but they're currently written in such different ways that it's not trivial to make them match. Filed a tech debt ticket: https://broadworkbench.atlassian.net/browse/AJ-1471.